File: src/main/java/com/warrantybee/api/enumerations/LogLevel.java
package com.warrantybee.api.enumerations;

/**
 * Represents the different levels of logging severity
 * used throughout the application.
 */
public enum LogLevel {

    /**
     * Informational messages that highlight
     * the normal progress of the application.
     */
    INFO,

    /**
     * Indicates a potentially harmful situation
     * or something that may require attention but
     * is not an error.
     */
    WARN,

    /**
     * Represents error events that may still allow
     * the application to continue running.
     */
    ERROR,

    /**
     * Fine-grained informational events useful
     * for debugging and development.
     */
    DEBUG
}

File: src/WarrantyBee.Domain/Enums/LogLevel.cs
namespace WarrantyBee.Domain.Enums;

/// <summary>
/// Specifies the severity level of a log message.
/// </summary>
public enum LogLevel
{
    /// <summary>
    /// Informational messages.
    /// </summary>
    Info,
    /// <summary>
    /// Warning messages indicating a potential issue.
    /// </summary>
    Warn,
    /// <summary>
    /// Error messages indicating a failure.
    /// </summary>
    Error,
    /// <summary>
    /// Diagnostic messages for debugging purposes.
    /// </summary>
    Debug
}
